semantic-ast/src/AST/Traversable1/Class.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+-- | This module defines the 'Traversable1' class and its generic derivation using 'GTraversable1'. Note that any changes to this file will require recompilation of all of the AST modules, which is quite expensive; thus, most additions should be made in "AST.Traversable1" instead, and that that module should not be imported by the AST modules.
+module AST.Traversable1.Class
```

Comment:
Splitting the `Traversable1`-relevant definitions up between `AST.Traversable1` & `AST.Traversable1.Class` like this means that `AST.GenerateSyntax`, and thus the lang-specific AST modules, can import this module, which should hopefully be very stable, while changes can be made to the other one. Consumers of the API should use the `AST.Traversable1` module, which re-exports this one and adds a few new definitions.

This is intended to minimize expensive rebuilding of the AST modules.